### PR TITLE
Convert tests to pytests

### PR DESCRIPTION
**Pull request by UkuLoskit**
Part of https://github.com/commaai/openpilot/issues/32941

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -22,7 +22,7 @@ jobs:
     - name: Build opendbc
       run: ${{ env.RUN }} "cd ../ && scons -j$(nproc) --minimal"
     - name: Unit tests
-      run: ${{ env.RUN }} "pytest"
+      run: ${{ env.RUN }} "pytest -n auto"
```
Comment (sshane) [inline .github/workflows/tests.yml:25]: openpilot uses -n logical for hyperthreading I believe

File: can/tests/test_define.py [hunk 2]
```diff
@@ -1,35 +1,27 @@
...
   def test_civic(self):
 
     dbc_file = "honda_civic_touring_2016_can_generated"
     defs = CANDefine(dbc_file)
 
-    self.assertDictEqual(defs.dv[399], defs.dv['STEER_STATUS'])
-    self.assertDictEqual(defs.dv[399],
-                         {'STEER_STATUS':
+    assert defs.dv[399] == defs.dv['STEER_STATUS']
+    assert defs.dv[399] == \
```
Comment (sshane) [inline can/tests/test_define.py:12]: are parenthesis more pretty here?

File: can/tests/test_define.py [hunk 3]
```diff
@@ -1,35 +1,27 @@
...
+                         {'STEER_STATUS': \
```
Comment (sshane) [inline can/tests/test_define.py:13]:
```suggestion
                         {'STEER_STATUS':
```

I don't think you need backslash here?

File: can/tests/test_define.py [hunk 4]
```diff
@@ -1,35 +1,27 @@
...
                           {7: 'PERMANENT_FAULT',
                            6: 'TMP_FAULT',
                            5: 'FAULT_1',
                            4: 'NO_TORQUE_ALERT_2',
                            3: 'LOW_SPEED_LOCKOUT',
                            2: 'NO_TORQUE_ALERT_1',
-                           0: 'NORMAL'}
+                           0: 'NORMAL'} \
```
Comment (sshane) [inline can/tests/test_define.py:20]:
```suggestion
                           0: 'NORMAL'}
```

File: .github/workflows/tests.yml (hunk 1 above)
Comment (maxime-desroches) [reply .github/workflows/tests.yml:25]: Normally it uses auto: https://github.com/commaai/openpilot/blob/master/pyproject.toml#L139
but in CI we use logical: https://github.com/commaai/openpilot/blob/master/.github/workflows/selfdrive_tests.yaml#L30

**Pull request merged**